### DISCUSSION TITLE
Invalid type for argument in function call. Invalid implicit conversion from uint256 to address requested in my "emit RequestedRaffleWinner(requestId);"

### DISCUSSION THREAD

**Discussion by yusuphoo** [Cyfrin/foundry-full-course-cu#1375] [Q&A] [answered]
function performUpkeep(bytes calldata /* performData */) external {
        // check to see if time has passed
        (bool upKeepNeeded, ) = checkUpKeep("");
        if (!upKeepNeeded) {
            revert Raffle_upkeepNotNeeded(
                address(this).balance,
                s_players.length,
                uint256(s_raffleState)
            );
        }
        //check to see if enough time has passed.
        s_raffleState = RaffleState.CALCULATING;
        uint256 requestId = i_vrfCoordinator.requestRandomWords(
            i_gasLane, // Gas Lane
            i_sunbscribtionId,
            i_callbackGasLimit,
            REQUEST_CONFIRMATIONS,
            NUM_WORDS
        );
        emit RequestedRaffleWinner(requestId);
    }

**Comment by yusuphoo** [comment]
This is the same as Patrick's code but it still shows red underlined error. I don't why.

**Comment by Niferu** [comment, marked as answer]
Hello @yusuphoo 

Check if you have declared that event same as below:

```solidity
event RequestedRaffleWinner(uint256 indexed requestId);
```

**Reply by yusuphoo** [reply]
This solved the issue right away. i declared it as "event RequestedRaffleWinner(Address indexed requestId);" Be blessed boss.
